File: day_02/day_02_p2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, _ := os.Open("./puzzle_input.txt")
	defer file.Close()

	scanner := bufio.NewScanner(file)

	x, y, aim := 0, 0, 0

	for scanner.Scan() {
		movement := strings.Split(scanner.Text(), " ")
		direction := movement[0]
		units, _ := strconv.Atoi(movement[1])

		switch direction {
		case "forward":
			x += units
			y += aim * units
		case "down":
			aim += units
		case "up":
			aim -= units
		}
	}

	fmt.Printf("Final position %v\n", x*y)
}

File: day_02/day_02_p2.R
input <- read.table('puzzle_input.txt')
input$V2[input$V1 == "up"] <- input$V2[input$V1 == "up"] * -1

depth <- 0
aim <- 0

for (ii in seq_along(input$V2)) { 
  if (input$V1[ii] == "forward") {
    depth <- depth + (aim * input$V2[ii])
  } else {
    aim <- aim + input$V2[ii]
  }
}

print(depth * sum(input$V2[input$V1 == 'forward']))
